src/components/Locations.js

```diff
@@ -0,0 +1,98 @@
+import { Component } from "react";
+import "../App.css";
+
+class Locations extends Component{
+    constructor(){
+        super();
+
+        this.state={
+            locations:[],
+            display:true,
+            locationName:"",
+            locationClimate:"",
+            locationTerrain:"",
+            buttonText: "Show Locations"
+        }
+    }
+  
+    fetchLocations=()=>{
+        fetch("https://ghibliapi.herokuapp.com/locations")
+            .then(res=>res.json())
+            .then ((data)=>{
+                this.setState({
+                    locations: data,
+            })
+        })
+    }
+
+    componentDidMount(){
+        this.fetchLocations();
+    }
+
+handleClick=(e)=>{
+    e.preventDefault();
+    let name = this.state.locations.map((location)=>location.name).join(", ");
+    let climate = this.state.locations.map((location)=>location.climate).join(", ");
+    let terrain = this.state.locations.map((location)=>location.terrain).join(", ");
+    this.setState({
+        locationName: name,
+        locationClimate: climate,
+        locationTerrain: terrain,
+        buttonText: "Hide Locations",
+
+    })
+}
```

Comment:
```suggestion
handleClick=()=>{
    this.setState({
    display: !this.state.display //<-toggles display between true and false
    })
}
```
1) you don't need `e.preventDefault()` because this is an event handler for a button. We usually use `e.preventDefault()` on **forms** because their default behavior is to cause a page refresh, and calling `e.preventDefault()` stops that from happening.

2) Clicking the button should just toggle `this.state.display` between **true** and **false**. 
     **!true** === **false** and **!false** === **true**